mlrun/serving/states.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from copy import deepcopy
+
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+import requests
+
+from ..model import ModelObj, ObjectDict
+from ..utils import create_class
+
+
+_task_state_fields = ["kind", "class_name", "class_args", "handler"]
+
+
+def model_endpoint(class_name, model_path, handler=None, **class_args):
+    class_args = deepcopy(class_args)
+    class_args["model_path"] = model_path
+    return ServingTaskState(class_name, class_args, handler=handler)
```

Comment:
this is creating a task state specific for model_endpoint and the next is specific to remote_endpoint its a way to decouple/abstract the details from the client code, its a class wrapper i'm not sure why `create_` is needed, if u insist we can change to `new_`

Reply:
1. function names tend to have verbs decribe what they're doing, look at all of your code `add_model` `get_class` `init_object` why didn't you call them `model` `class` `object` ?
2. you're calling here to a function `model_endpoint`. 
If `endpoint` was some kind of well defined term in the code... niha, but it's not, search for endpoint and you'll find only the obvious usage - a URL, so I don't know why you think someone will ever understand/use a function called `model_endpoint` that returns a `ServingTaskState` class instance.
saying again, I've really tried to get into what's going on in this PR, but you added tons of classes, you're switching terms each and everyline (for "abstraction"), I accept that maybe I don't see the whole picture, but since you didn't add a description or anything that helps understanding it, I'm helpless

Reply:
changed to new_